### PR TITLE
Fix Graph extract marker with const

### PR DESCRIPTION
**Pull request by zhen8838**
1. When extracting the maker with const Eclass. the `Linq.MinBy` method will return the const by default.
2. Add the `MinByWithMarker/MinByWithOutMarker` method to select `minCodeEnode` correctly.
截屏2023-01-09 10 33 58
3. add `FoldConv2DMulAdd` rule for merge bn before conv2d.
4. make the `Uninitialized` cost return zero.

### CODE REVIEW

File: src/Nncase.Core/IR/Marker.cs [hunk 1]
```diff
@@ -28,16 +28,10 @@ public class MixQuantInfo
...
 public sealed record Marker(string Name, Expr Target, Expr Attribute) : Expr
 {
-    public MixQuantInfo MixQuantInfo;
-
     /// <summary>
-    /// Gets Target.
+    /// Gets or set the mix quant info.
     /// </summary>
-    /// <returns></returns>
-    public Expr MarkerTarget()
```
Comment (aaltonenzhang) [inline src/Nncase.Core/IR/Marker.cs:37]: remove this now will make 510/230 compile fail.

File: src/Nncase.Transform/Rules/Neutral/FoldConv2DMulAdd.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            return null;
+        }
+
+        int ic = weights.Shape[1].FixedValue;
+        if (mulConst.Length != ic || addConst.Length != ic)
+        {
+            return null;
+        }
+
+        var new_weights = IR.F.Math.Mul(weights, Reshape(mulConst, new[] { 1, ic, 1, 1 }));
```
Comment (FusionBolt) [inline src/Nncase.Transform/Rules/Neutral/FoldConv2DMulAdd.cs:106]: Use camel naming.

File: src/Nncase.Core/IR/Marker.cs (hunk 1 above)
Comment (zhen8838) [reply src/Nncase.Core/IR/Marker.cs:37]: I will fix it.

File: src/Nncase.Transform/Rules/Neutral/FoldConv2DMulAdd.cs (hunk 2 above)
Comment (zhen8838) [reply src/Nncase.Transform/Rules/Neutral/FoldConv2DMulAdd.cs:106]: done

**Review by sunnycase — APPROVED** [review]

**Pull request merged**